ksql-functional-tests/src/test/resources/query-validation-tests/delimited-value-format.json

```diff
@@ -0,0 +1,222 @@
+{
+  "comments": [
+    "When using value_format DELIMITED, we can define VALUE_DELIMITER as custom character."
+  ],
+  "tests": [
+    {
+      "name": "select delimited value_format",
+      "statements": [
+        "CREATE STREAM TEST (ID bigint, NAME varchar, VALUE integer) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE STREAM S2 as SELECT id, name, value FROM test;"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": "0,zero,0", "timestamp": 0},
+        {"topic": "test_topic", "key": 0, "value": "0,100,100", "timestamp": 0},
+        {"topic": "test_topic", "key": 100, "value": "100,100,500", "timestamp": 0},
+        {"topic": "test_topic", "key": 100, "value": "100,100,100", "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": "0,zero,0", "timestamp": 0},
+        {"topic": "S2", "key": 0, "value": "0,100,100", "timestamp": 0},
+        {"topic": "S2", "key": 100, "value": "100,100,500", "timestamp": 0},
+        {"topic": "S2", "key": 100, "value": "100,100,100", "timestamp": 0}
+      ]
+    },
+    {
+      "name": "validate value_delimiter to be single character",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter='<~>');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter must be a single character, 'TAB' or 'SPACE'.\nExample valid value: ';'"
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "validate delimeter is not empty",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter='');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter cannot be empty or whitespace."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "validate delimeter is not a space",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter=' ');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter cannot be empty or whitespace."
```

Comment:
Would be good if this included "If you intended the delimiter to be a space or a tab then you should use `'SPACE'` or `'TAB'`, respectively."